lib/internal/async_hooks.js

```diff
@@ -103,6 +105,26 @@ const emitDestroyNative = emitHookFactory(destroy_symbol, 'emitDestroyNative');
 const emitPromiseResolveNative =
     emitHookFactory(promise_resolve_symbol, 'emitPromiseResolveNative');
 
+function callbackTrampoline(asyncId, cb, domain_cb, ...args) {
+  if (hasHooks(kBefore))
+    emitBeforeNative(asyncId);
+
+  let result;
+  if (typeof domain_cb === 'function') {
+    args.unshift(cb);
+    result = ReflectApply(domain_cb, this, args);
+  } else {
+    result = ReflectApply(cb, this, args);
```

Comment:
Is there a pref gap between this and calling `apply` ?

Reply:
No idea, but it's safer to use `ReflectApply` here because someone nefarious could redefine `Function.prototype.apply` to do something else. `ReflectApply` doesn't rely on the prototype to call the function. It seems to be the accepted standard of core to use it, so I just did what everywhere else seems to be doing. 🤷

Reply:
I think older V8 versions have a perf gap? On v14.x this should be the best way, though :+1:

Reply:
I just did a test with stable Chrome and got this:

```
> foo.apply(null, [1, 2, 3])
66,498,480 ops/s ±0.4%
fastest
> Reflect.apply(foo, null, [1, 2, 3])
66,325,587 ops/s ±0.57%
fastest
```

Looks like the difference is basically irrelevant. 👍